trunner/device.py

```diff
@@ -44,6 +64,110 @@ def run(self, test):
             self.serial.close()
 
 
+class GPIO:
+    def __init__(self, pin):
+        self.pin = pin
+
+        RPi.GPIO.setmode(RPi.GPIO.BCM)
+        RPi.GPIO.setwarnings(False)
+        RPi.GPIO.setup(self.pin, RPi.GPIO.OUT)
+
+    def high(self):
+        RPi.GPIO.output(self.pin, RPi.GPIO.HIGH)
+
+    def low(self):
+        RPi.GPIO.output(self.pin, RPi.GPIO.LOW)
+
+
+class IMXRT106xRunner(DeviceRunner):
+    SDP = 'plo-ram-armv7m7-imxrt106x.sdp'
+    IMAGE = 'phoenix-armv7m7-imxrt106x.disk'
+
+    def __init__(self, port, phoenixd_port='/dev/ttyUSB0'):
+        super().__init__(port)
+        self.phoenixd_port = phoenixd_port
+        self.serial_downloader = False
+
+        self.reset_gpio = GPIO(17)
+        self.reset_gpio.high()
+
+        self.boot_gpio = GPIO(4)
+
+    def reset(self):
+        self.reset_gpio.low()
+        time.sleep(0.050)
+        self.reset_gpio.high()
+
+    def boot(self):
+        if self.serial_downloader:
+            self.boot_gpio.low()
+        else:
+            self.boot_gpio.high()
+
+        self.reset()
+
+    def flash(self):
+        self.serial_downloader = True
+        self.boot()
+
+        logging.info("psu run!\n")
```

Comment:
I'm wondering about detecting if we're running inside github actions and styling the output with specific "tags" - I'm thinking about `::group::` to create collapsible debug logs (https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#grouping-log-lines) - like the current use case when we're flushing the device and are interested in the logs only if it failed.

Just an idea to experiment with in the future.

Reply:
Wow, I didn't know we can style output in github actions. I noted it for the future.

Reply:
yeah, we can also emit warnings, errors and other things :)